### ISSUE TITLE
mesh.smoothed fails on some meshes

### ISSUE THREAD

**Issue by FreakTheMighty** [mikedh/trimesh#591]
I'm finding that for some meshes, the `smoothed` function fails. I've attached a sample GLB, a simple script, and the traceback I see. 

I'm a little confused on the name of the method, cause I'm actually using it to break up faces that should not rendered with smooth shading. That is, my model contains many right angles, and the `smoothing` method seems to break faces by angle, which prevents it from rending with smoothed normals.

[trysmooth.glb.zip](https://github.com/mikedh/trimesh/files/3805538/trysmooth.glb.zip)

```python
import trimesh

scene = trimesh.load_mesh('./trysmooth.glb')
for mesh in scene.geometry:
    scene.geometry[mesh].smoothed()
```

```bash
  File "./trysmooth.py", line 4, in <module>
    scene.geometry['geometry_0'].smoothed()
  File "/Users/jesse/Code/IfcTiler/env/lib/python3.7/site-packages/trimesh-3.2.39-py3.7.egg/trimesh/constants.py", line 136, in timed
    result = method(*args, **kwargs)
  File "/Users/jesse/Code/IfcTiler/env/lib/python3.7/site-packages/trimesh-3.2.39-py3.7.egg/trimesh/base.py", line 1930, in smoothed
    facet_minlen=facet_minlen)
  File "/Users/jesse/Code/IfcTiler/env/lib/python3.7/site-packages/trimesh-3.2.39-py3.7.egg/trimesh/graph.py", line 783, in smoothed
    facets = [f for f in mesh.facets
  File "/Users/jesse/Code/IfcTiler/env/lib/python3.7/site-packages/trimesh-3.2.39-py3.7.egg/trimesh/caching.py", line 101, in get_cached
    value = function(*args, **kwargs)
  File "/Users/jesse/Code/IfcTiler/env/lib/python3.7/site-packages/trimesh-3.2.39-py3.7.egg/trimesh/base.py", line 1622, in facets
    facets = graph.facets(self)
  File "/Users/jesse/Code/IfcTiler/env/lib/python3.7/site-packages/trimesh-3.2.39-py3.7.egg/trimesh/graph.py", line 269, in facets
    radii = mesh.face_adjacency_radius
  File "/Users/jesse/Code/IfcTiler/env/lib/python3.7/site-packages/trimesh-3.2.39-py3.7.egg/trimesh/caching.py", line 101, in get_cached
    value = function(*args, **kwargs)
  File "/Users/jesse/Code/IfcTiler/env/lib/python3.7/site-packages/trimesh-3.2.39-py3.7.egg/trimesh/base.py", line 1411, in face_adjacency_radius
    radii, span = graph.face_adjacency_radius(mesh=self)
  File "/Users/jesse/Code/IfcTiler/env/lib/python3.7/site-packages/trimesh-3.2.39-py3.7.egg/trimesh/graph.py", line 173, in face_adjacency_radius
    point_pairs = mesh.vertices[mesh.face_adjacency_unshared]
  File "/Users/jesse/Code/IfcTiler/env/lib/python3.7/site-packages/trimesh-3.2.39-py3.7.egg/trimesh/caching.py", line 101, in get_cached
    value = function(*args, **kwargs)
  File "/Users/jesse/Code/IfcTiler/env/lib/python3.7/site-packages/trimesh-3.2.39-py3.7.egg/trimesh/base.py", line 1398, in face_adjacency_unshared
    vid_unshared = graph.face_adjacency_unshared(self)
  File "/Users/jesse/Code/IfcTiler/env/lib/python3.7/site-packages/trimesh-3.2.39-py3.7.egg/trimesh/graph.py", line 141, in face_adjacency_unshared
    vid_unshared[:, i] = faces[np.logical_not(shared)]
ValueError: could not broadcast input array from shape (19590) into shape (19599)
```

**Comment by mikedh** [comment]
Thanks for the report! That looks like an index issue due to some degenerate faces, I'll push a fix.

With regards to smoothing, it is only a visual thing where it disconnects faces with an angle above a threshold so they look better when rendered with openGL. If you have strong opinions about what is visually connected/smooth, you're probably off modifying the mesh yourself and then visualizing with `mesh.show(smooth=False)`. 


EG with a watertight mesh and no smoothing you see facets:
![nosmooth](https://user-images.githubusercontent.com/1403074/68160909-bd374e00-ff22-11e9-857a-47a60886ff10.png)

Versus smooth looking curved areas and disconnected sharp edges:
![smooth](https://user-images.githubusercontent.com/1403074/68160945-d63fff00-ff22-11e9-9153-4feaf75184f5.png)

**Comment by mikedh** [comment]
Should be fixed by #592, feel free to re-open if not. Thanks for the report!

**Comment by FreakTheMighty** [comment]
That worked, thank you!
